src/frontends/tensorflow/src/op/segment_sum.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "op_table.hpp"
+#include "openvino/opsets/opset9.hpp"
+
+using namespace std;
+using namespace ov;
+using namespace ov::opset9;
+
+namespace ov {
+namespace frontend {
+namespace tensorflow {
+namespace op {
+OutputVector translate_segment_sum_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"SegmentSum"});
+    auto data = node.get_input(0);
+    auto segment_ids = node.get_input(1);
+
+    // compute SegmentSum using EmbeddingSegmentSum
+    // for this prepare all the required inputs
+    auto indices_type = segment_ids.get_element_type();
+    // 1. compute a number of segments using segment_ids values
+    // do not forget that segment ids are counting from zero
+    auto reduction_axis = make_shared<Constant>(element::i32, Shape{1}, 0);
```

Comment:
Nitpick: I don't remember exactly about the agreement what better to use in our own code i32 or i64. I guess i64 but I'm not so sure

Reply:
i32 is good enough and less memory.